server/prometheus.js

```diff
@@ -50,14 +26,108 @@ class Prometheus {
         };
     }
 
+    /**
+     * Initialize Prometheus metrics, and add all available tags as possible labels.
+     * This should be called once at the start of the application.
+     * New tags will NOT be added dynamically, a restart is sadly required to add new tags to the metrics.
+     * Existing tags added to monitors will be updated automatically.
+     * @returns {Promise<void>}
+     */
+    static async init() {
+        // Add all available tags as possible labels,
+        // and use Set to remove possible duplicates (for when multiple tags contain non-ascii characters, and thus are sanitized to the same label)
+        const tags = new Set((await R.findAll("tag")).map((tag) => {
+            return Prometheus.sanitizeForPrometheus(tag.name);
+        }).filter((tagName) => {
+            return tagName !== "";
+        }));
+
+        const commonLabels = [
+            ...tags,
+            "monitor_id",
+            "monitor_name",
+            "monitor_type",
+            "monitor_url",
+            "monitor_hostname",
+            "monitor_port",
+        ];
+
+        monitorCertDaysRemaining = new PrometheusClient.Gauge({
+            name: "monitor_cert_days_remaining",
+            help: "The number of days remaining until the certificate expires",
+            labelNames: commonLabels
+        });
+
+        monitorCertIsValid = new PrometheusClient.Gauge({
+            name: "monitor_cert_is_valid",
+            help: "Is the certificate still valid? (1 = Yes, 0= No)",
+            labelNames: commonLabels
+        });
+
+        monitorResponseTime = new PrometheusClient.Gauge({
+            name: "monitor_response_time",
+            help: "Monitor Response Time (ms)",
+            labelNames: commonLabels
+        });
+
+        monitorStatus = new PrometheusClient.Gauge({
+            name: "monitor_status",
+            help: "Monitor Status (1 = UP, 0= DOWN, 2= PENDING, 3= MAINTENANCE)",
+            labelNames: commonLabels
+        });
+    }
+
+    /**
+     * Sanitize a string to ensure it can be used as a Prometheus label or value.
+     * See https://github.com/louislam/uptime-kuma/pull/4704#issuecomment-2366524692
+     * @param {string} text The text to sanitize
+     * @returns {string} The sanitized text
+     */
+    static sanitizeForPrometheus(text) {
+        text = text.replace(/[^a-zA-Z0-9_]/g, "");
+        text = text.replace(/^[^a-zA-Z_]+/, "");
+        return text.toLowerCase();
```

Comment:
Let's not change it this way

```suggestion
        return text;
```